Lib/_collections_abc.py

```diff
@@ -1161,3 +1161,13 @@ def __iadd__(self, values):
 
 MutableSequence.register(list)
 MutableSequence.register(bytearray)
+
+_deprecated_ByteString = globals().pop("ByteString")
+
+def __getattr__(attr):
+    if attr == "ByteString":
+        import warnings
+        warnings._deprecated("collections.abc.ByteString", remove=(3, 17))
+        globals()["ByteString"] = _deprecated_ByteString
```

Comment:
This means only the first time anyone does `from collections.abc import ByteString` in a program will trigger a warning, right? That seems potentially problematic: maybe the first time is in some dependency you don't control, so you ignore the warning, and then it happens again in your code and you miss it.

Reply:
Probably OK to stick with the standard format here, hopefully it's enough to get people to notice...